spec.bs

```diff
@@ -646,14 +646,22 @@ To <dfn>construct a pending fenced frame config</dfn> given an [=auction config=
 To <dfn>fill in a pending fenced frame config</dfn> given a [=fenced frame config=]
 |pendingConfig|, [=auction config=] |auctionConfig|, and [=leading bid info=] |winningBidInfo|:
 1. Let |winningBid| be |winningBidInfo|'s [=leading bid info/leading bid=].
-1. Set |pendingConfig|'s [=fenced frame config/mapped url=]'s [=mapped url/value=] to
-   |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
 1. Let |adSize| be |winningBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/size=].
+1. Let |concreteAdSize| be the result of TODO: resolving screen-relative sizes to pixel values for
+   |adSize|.
+1. Let |adWidthString| and |adHeightString| be the result of TODO: converting |concreteAdSize| to
+   strings.
+1. Let |sizeMacroMap| be the [=ordered map=] «[ "{%AD_WIDTH%}" → |adWidthString|,
+   "{%AD_HEIGHT%}" → |adHeightString|, "${AD_WIDTH}" → |adWidthString|,
+   "${AD_HEIGHT}" → |adHeightString|]»
+1. Set |pendingConfig|'s [=fenced frame config/mapped url=]'s [=mapped url/value=] to
+   the result of [=fencedframeutil/substituting macros=] with |sizeMacroMap| into |winningBid|'s
+   [=generated bid/ad descriptor=]'s [=ad descriptor/url=].
 1. If |adSize| is not null:
    1. Set |pendingConfig|'s [=fenced frame config/content size=] to a [=struct=] with the following
       [=struct/items=]:
      : [=content size/value=]
-     :: |adSize| TODO: Resolve screen-relative sizes and macros and cast this properly.
+     :: |adSize|, |concreteAdSize|. TODO: cast this properly

```

Comment:
Can you clarify what the remaining "cast this properly" means here? What is incorrect about this assignment as-is.

Reply:
The type of `[=fenced frame config/content size=]` isn't exactly final:

https://wicg.github.io/fenced-frame/#:~:text=TODO%3A%20Consider%20different%20numeric%20types%20for%20these%20members.

And there is the TODO above about resolving the sizes to pixel values. So I was mostly gesturing at the fact that there might need to be changes here when those things are finalized.

Reply:
Got it.

Reply:
Probably best to wrap the TODO in an XXX box, and also link to any related issues that are relevant to this, that will need to be followed-up on. Just a plain text TODO is really easy to lose track of. Even better would be an `Issue: Foo` box, since those all accumulate at the bottom of the spec and are uniquely linkable.

Reply:
is the |concreteAdSize| going to replace the |adSize|, or both will be passed to [=fenced frame config/content size=]? I think the TODO needs to be more clear about what needs to be done. And the current way of passing two values to [=content size/value=] is not clear what it does.

Reply:
Clarified that this is about casting |concreteAdSize| into the format for the fenced frame config content size field, which has not been completely finalized.